### PR TITLE
feat: Support broker hostname assignment of external PoW

### PR DESCRIPTION
**Pull request by marktwtn**
# Description

DO NOT MERGE

For assigning the broker hostname when using the external PoW library.
The broker is in charge of distributing the PoW calculation to other faster hardware.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

### CODE REVIEW

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 1]
```diff
@@ -95,6 +95,7 @@
 
     //PearlDiver
     protected int powThreads = Defaults.POW_THREADS;
+    protected String brokerIP = Defaults.BROKER_IP;
```
Comment (jserv) [inline src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: If you are referring to IPv4 address, it is indeed a 32-bit integer. If you intended to mention `host` or `hostname`, you should revise the usage for these terms.

Comment (marktwtn) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Both of the `hostname` and the `IP address` expression should be acceptable.
The data type should use `String`.

The variable naming might be confusing.
I will change it to `brokerHostname` and update the description that both of the `hostname` and the `IP address` are acceptable.

Comment (marktwtn) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Done.

**Comment by jserv** [comment]
Fix CI regression.

**Comment by marktwtn** [comment]
> Fix CI regression.

The bug is caused by the **Travis CI**.

When it uses **OpenJDK8**, it would use **Java 8 runtime environment** and **javac 9**.
The version mismatching would make the rocksdb fail.

I will create another pull request to add the **OpenJDK8** testing into the `allow failure` case.

Bug reference:
https://github.com/travis-ci/travis-ci/issues/9867

**Comment by marktwtn** [comment]
Blocked by #8.

**Comment by marktwtn** [comment]
The #8 has been merged and CI are passed.
The pull request can be merged now.

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 2]
```diff
@@ -95,6 +95,7 @@
 
     //PearlDiver
     protected int powThreads = Defaults.POW_THREADS;
+    protected String brokerHostname = Defaults.BROKER_HOSTNAME;
```
Comment (jserv) [inline src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: How about `brokerHost`?

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 3]
```diff
@@ -891,6 +903,7 @@ protected void setPowThreads(int powThreads) {
 
         //PearlDiver
         int POW_THREADS = 0;
+        String BROKER_HOSTNAME = "localhost";
```
Comment (jserv) [inline src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Should the default hostname localhost?

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java (hunk 2 above)
Comment (marktwtn) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Sounds good.

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java (hunk 3 above)
Comment (marktwtn) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Or maybe we can leave the default value as an empty `String`.
I am fine with both option, but I prefer to set the default value as `localhost`.

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java (hunk 2 above)
Comment (marktwtn) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Done.

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java (hunk 3 above)
Comment (marktwtn) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: I decide to use the `localhost` as default value.

**Pull request merged**